src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(health, name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const decreaseHealthBy = 50;
+
+    if (!animal.hidden && !(animal instanceof Carnivore)) {
```

Comment:
does this make sense to firstly check whether the animal is hidden? What if we pass Carnivore there? Typescript would bring you an error in such case

Reply:
also, why you don't just check whether the animal is instanceof Herbivore, but you check whether they are not instanceof Carnivore?